data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/beacon/ValidatorResponse.java

```diff
@@ -60,14 +63,18 @@ public ValidatorResponse(
   }
 
   public static ValidatorResponse fromState(final BeaconState state, final Integer index) {
-    tech.pegasys.teku.datastructures.state.Validator validatorInternal =
-        state.getValidators().get(index);
-    final UInt64 current_epoch = compute_epoch_at_slot(state.getSlot());
-    return new ValidatorResponse(
-        UInt64.valueOf(index),
-        state.getBalances().get(index),
-        getValidatorStatus(current_epoch, validatorInternal),
-        new Validator(validatorInternal));
+    try {
+      tech.pegasys.teku.datastructures.state.Validator validatorInternal =
+          state.getValidators().get(index);
+      final UInt64 current_epoch = compute_epoch_at_slot(state.getSlot());
+      return new ValidatorResponse(
+          UInt64.valueOf(index),
+          state.getBalances().get(index),
+          getValidatorStatus(current_epoch, validatorInternal),
+          new Validator(validatorInternal));
+    } catch (IndexOutOfBoundsException ex) {
+      return ValidatorResponse.EMPTY;
```

Comment:
Would be better to check the index is valid ahead of time rather than catching the `IndexOutOfBoundsException`.